firmware/quadruna/BMS/src/app/app_accumulator.c

```diff
@@ -458,115 +518,97 @@ void app_accumulator_broadcast(void)
 
     // Calculate and broadcast pack power.
     const float available_power =
-        MIN(app_math_linearDerating(max_cell_temp, MAX_POWER_LIMIT_W, CELL_ROLL_OFF_TEMP_DEGC, CELL_FULLY_DERATED_TEMP),
+        MIN(app_math_linearDerating(
+                temp_stats.max_temp_cell.temp, MAX_POWER_LIMIT_W, CELL_ROLL_OFF_TEMP_DEGC, CELL_FULLY_DERATED_TEMP),
             MAX_POWER_LIMIT_W);
 
     app_canTx_BMS_AvailablePower_set((uint32_t)available_power);
 }
 
 bool app_accumulator_checkFaults(void)
 {
-    uint8_t throwaway_segment = 0U;
-    uint8_t throwaway_loc     = 0U;
-
-    float max_allowable_cell_temp = MAX_CELL_DISCHARGE_TEMP_DEGC;
-    float min_allowable_cell_temp = MIN_CELL_DISCHARGE_TEMP_DEGC;
-
     // if we are charging, max cell temp is 45C not 60C
-    if (app_canRx_BRUSA_IsConnected_get())
-    {
-        max_allowable_cell_temp = MAX_CELL_CHARGE_TEMP_DEGC;
-        min_allowable_cell_temp = MIN_CELL_CHARGE_TEMP_DEGC;
-    }
-
-    const State *current_state = app_stateMachine_getCurrentState();
+    const float max_allowable_cell_temp =
```

Comment:
Reminder that regen counts as charging, and should be limited/disabled at high cell temperatures